### PR TITLE
Signed cookies

### PR DESCRIPTION
**Pull request by marioapbrito**
Add a middleware which sets CloudFront signed cookies in your response.

### CODE REVIEW

File: cognito_code_grant/signed_cookies.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+        backend=default_backend()
+    )
+
+
+def add_signed_cookies(response):
+    # Generate signed cookies and add them to the response.
+    # The idea is to generate the cookies at each request and have them relatively short lived.
+    try:
+        if getattr(settings, 'SIGNED_COOKIES', False):
+            cookies = generate_signed_cookies()
```
Comment (panovitch) [inline cognito_code_grant/signed_cookies.py]: Shouldnt we also pass in the target resource we are getting access to (as in bucket)? Probably could also be taken from django settings

Comment (marioapbrito) [reply cognito_code_grant/signed_cookies.py]: Ah yes, indeed. Right now we're granting access to * , which pretty much is all buckets. I'll tweak the code a bit. Thx!

File: cognito_code_grant/signed_cookies.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        self.domain = None
+        try:
+            download_dist = connection.get_distribution(Id=download_dist_id)
+            if cname and download_dist.get('Distribution', {}).get('AliasICPRecordals', {}):
+                self.domain = download_dist['Distribution']['AliasICPRecordals'][0]['CNAME'] #use first cname if defined
+            else:
+                self.domain = download_dist.domain_name
+            self.download_dist = download_dist
+        except Exception as ex:
+            logging.error(ex)
```
Comment (andreparames) [inline cognito_code_grant/signed_cookies.py]: Should we swallow this exception? Seems like if this fails, the later call to `get_http_resource_url` will just generate an invalid result (something like `http://None/...`).

Comment (marioapbrito) [reply cognito_code_grant/signed_cookies.py]: I might remove this entire function and just pass the domain via env var. Gonna run a few tests and see if it suits our needs.

**Pull request merged**